agent/agent.go

```diff
@@ -369,6 +382,9 @@ func (a *Agent) handleSessionMessage(ctx context.Context, message *api.SessionMe
 			}
 		}
 	}
+	if len(message.RootCA) > 0 && digest.FromBytes(message.RootCA).String() != currentTLSInfo.TrustRootHash {
+		a.config.NotifyRootChange <- message.RootCA
```

Comment:
I believe this channel send is okay. The goroutine in `node` that receives from it won't return until the agent shuts down.